pkg/providers/instance/instance.go

```diff
@@ -440,10 +444,16 @@ func (p *Provider) launchInstance(
 	return resp, instanceType, nil
 }
 
+// isSKUNotAvailable - to be moved to azure-sdk-for-go-extensions
+func isSKUNotAvailable(err error) bool {
+	azErr := sdkerrors.IsResponseError(err)
+	return azErr != nil && azErr.ErrorCode == "SkuNotAvailable"
+}
```

Comment:
Can you share the arm call and the response we get back here? Just so others(Me) can be aware of the full shape of the SkuNotAvailable Subcode?

Reply:
> Code="SkuNotAvailable" Message="The requested size for resource '(redacted)' is currently not available in location '(redacted)' zones '1,2,3' for subscription '(redacted)'. Please try another size or deploy to a different location or zones. See https://aka.ms/azureskunotavailable for details."

Reply:
This actually makes me think that maybe we can be smarter about the zones here? Though I have not observed this with other than all zones in the region, it may well be possible ...

Reply:
Giving this some more thought, it seems we don't expect to observe SkuNotAvailable for regular instances once we introduce filtering based on SKU location restriction in https://github.com/Azure/karpenter/pull/35. Refocusing this to handle the remaining case of out-of-capacity for Spot, see updated comments in code.